src/org/traccar/database/NotificationManager.java

```diff
@@ -130,8 +130,11 @@ public Notification getUserNotificationByType(long userId, String type) {
     public void updateNotification(Notification notification) {
         Notification cachedNotification = getUserNotificationByType(notification.getUserId(), notification.getType());
         if (cachedNotification != null) {
-            if (!cachedNotification.getAttributes().equals(notification.getAttributes())) {
-                if (notification.getAttributes().isEmpty()) {
+            if (cachedNotification.getWeb() != notification.getWeb()
+                    || cachedNotification.getMail() != notification.getMail()
+                    || !cachedNotification.getAttributes().equals(notification.getAttributes())) {
```

Comment:
Let discuss it.
Actually we stop using attributes.
We need just decide if we need store notification like?
````
web=false
mail=false
attributes={ something }
````
 It can only happen if somebody use attributes for own tasks.